spec/ruby/core/enumerable/slice_when_spec.rb

```diff
@@ -0,0 +1,36 @@
+require File.expand_path('../../../spec_helper', __FILE__)
+require File.expand_path('../fixtures/classes', __FILE__)
+
+describe "Enumerable#slice_when" do
+  before :each do
+    ary = [10, 9, 7, 6, 4, 3, 2, 1]
+    @enum = EnumerableSpecs::Numerous.new *ary
+    @result = @enum.slice_when { |i, j| i - 1 != j }
+    @enum_length = ary.length
+  end
+
+  context "when given a block" do
+    it "returns an enumerator" do
+      @result.should be_an_instance_of(enumerator_class)
+    end
+
+    it "splits chunks between adjacent elements i and j where the block returns true" do
+      @result.to_a.should == [[10, 9], [7, 6], [4, 3, 2, 1]]
+    end
+
+    it "calls the block the length of the receiver enumerable minus one times" do
+      times_called = 0
+      @enum.slice_when do
+        times_called += 1
+        false
```

Comment:
Alright, I think it's a bad idea to use `false`. I've decided to stick to my `i - 1 != j` example in my latest commit.